src/main/java/com/examenes/vault/service/VaultService.java

```diff
@@ -27,7 +29,17 @@
     public Employee create(Employee employee) {
         EmployeeModel employeeModel = EmployeeMapper.toModel(employee);
 
-        //
+        List<EmployeeModel> listEmp = employeeRepository.findAll()
+                .stream().filter(emp -> emp.getDepartment().getDepartmentId() == employeeModel.getDepartment().getDepartmentId()).collect(Collectors.toList());
+        Long sumSalary = 0L;
+        for (EmployeeModel e : listEmp) {
+            sumSalary += e.getSalary();
+        }
+        Long avgSalary = sumSalary / listEmp.size();
+        Date today = Date.from(Instant.now());
+        if ((Calendar.DAY_OF_MONTH < 14 && avgSalary > 1000) || (Calendar.DAY_OF_MONTH > 14 && avgSalary > 1500)) {
+            throw new RequestException("Denied", "denied", HttpStatus.CONFLICT.value());
```

Comment:
el conflict no si está del todo bien, según la docu:

> Conflicts are most likely to occur in response to a PUT request. For example, you may get a 409 response when uploading a file which is older than the one already on the server resulting in a version control conflict.

Capaz mandaría un bad request, y además, no estás siendo denegado, sino que no cumpliste con una regla de negocio. Así que pondría un mensaje que ayude a reconocer la validación, por ejemplo "avg salary is xxx or date is not xxx"

Reply:
Si, es un poco complicado, porque en si el 400 sería para un request mal armado, pero acá lo armó bien, solo que no cumple con una regla de negocio. Yo le dije que ponga 409 pero tambien tenia mis dudas, quizas hay otro, de los 42x que sirva para este caso.

Reply:
Creo que el que mas se ajustaba era 422, pero nadie los conoce esos

Reply:
Estos son por ejemplo los códigos que usa google:
https://google.aip.dev/193
https://github.com/googleapis/api-common-protos/blob/master/google/rpc/code.proto

Reply:
Si, para mi es un 400, igual tampoco veo mal el 409. Aca podes tirar una excepcion de negocio sin atarla a un http status, y en el handler de excepciones la traducis a un bad request.